aspnetcore/signalr/troubleshoot.md

```diff
@@ -0,0 +1,91 @@
+---
+title: ASP.NET Core SignalR connection troubleshooting
+author: 
+description: ASP.NET Core SignalR connection troubleshooting.
+monikerRange: '>= aspnetcore-2.1'
+ms.author: 
+ms.custom: mvc
+ms.date: 04/08/2020
+no-loc: ["ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+uid: signalr/troubleshoot
+---
+# Troubleshoot connection errors
+
+This section provides help with errors that can occur when trying to establish a connection to a ASP.NET Core SignalR hub.
+
+### Response code 404
+
+When using WebSockets and `skipNegotiation = true`
+```log
+WebSocket connection to 'wss://xxx/HubName' failed: Error during WebSocket handshake: Unexpected response code: 404
+```
+
+* If the connection uses the ID and takes too long to send a request to the server after the negotiate, the server:
+
+  * Deletes the ID.
+  * Returns a 404.
+* When using multiple servers without sticky sessions, the connection can start on one server and then switch to another server. The other server is not aware of the previous connection.
+* Verify the client is connecting to the correct endpoint. For example, the server is hosted at `http://127.0.0.1:5000/hub/myHub` and client is trying to connect to `http://127.0.0.1:5000/myHub`.
+
+### Response code 400 or 503
+
+For the following error:
+
+```log
+WebSocket connection to 'wss://xxx/HubName' failed: Error during WebSocket handshake: Unexpected response code: 400
+
+Error: Failed to start the connection: Error: There was an error with the transport.
+```
+
+This error is usually caused by client using only WebSockets transport but the WebSockets protocol is not enabled on the server.
+
+### Response code 307
+
+When using WebSockets and `skipNegotiation = true`
+```log
+WebSocket connection to 'ws://xxx/HubName' failed: Error during WebSocket handshake: Unexpected response code: 307
+```
+
+This error can also happen during the negotiate request.
```

Comment:
Do any of the HTTP clients and/or WebSocket clients SignalR uses not follow 307 redirects? I just searched the aspnetcore repo for "SignalR 307" (without quotes) and came up with no reports of an issue like that.

Reply:
Found it: https://stackoverflow.com/questions/63437471/websocket-connection-to-ws-localhost5000-notificationhub-failed-error-duri. I'm still not sure that this could "happen during the negotiate request." Do you have evidence of that?

Reply:
Yes, I point out one possible cause below

Reply:
I don't see it. Why would a 307 break the negotiate request itself?

Reply:
Because we don't follow redirects

Reply:
Oh. Why? Because we assume it will break WebSockets?

Reply:
Hmm, I'm not sure. I thought it was because auth handlers often redirect instead of returning 401, but we set a header to try and avoid that...

Reply:
https://github.com/dotnet/aspnetcore/issues/27068